File: gear_generator/gear_svg.dart
import 'package:xml/xml.dart';
import 'gear_path.dart';
import 'package:meta/meta.dart';

class GearSvg {
  GearSvg({@required this.path, @required this.width, @required this.height});

  final GearPath path;
  final int width;
  final int height;

  @override
  String toString() {
    final builder = XmlBuilder();

    builder.processing('xml', 'version="1.0" encoding="UTF-8"');
    builder.element('svg', nest: () {
      builder.namespace("http://www.w3.org/2000/svg");
      builder.attribute("version", "1.1");
      builder.attribute("width", width);
      builder.attribute("height", height);
      builder.attribute("viewbox", "0 0 $width $height");
      builder.element('g', nest: () {
        builder.element('path', nest: () {
          builder.attribute('d', path.toString());
          builder.attribute('style',
              'fill:#000000; fill-opacity:0.3; stroke:#000000; stroke-width:0.25;');
        });
      });
    });

    return builder.buildDocument().toXmlString(pretty: true);
  }
}

File: gear_generator_2/src/gear_svg.ts
import { GearPath } from './gear_path';
import { create } from 'xmlbuilder2';

interface GearSvgConstructorParams {
  path: GearPath;
  width: number;
  height: number;
}

export class GearSvg {
  public params;

  constructor(params: GearSvgConstructorParams) {
    this.params = params;
  }
}

GearSvg.prototype.toString = function (): string {
  return create()
    .ele({
      'svg@http://www.w3.org/2000/svg': {
        '@version': '1.1',
        '@width': this.params.width,
        '@height': this.params.height,
        '@viewbox': `0 0 ${this.params.width} ${this.params.height}`,
        g: {
          path: {
            '@d': this.params.path.toString(),
            '@style':
              'fill:#000000; fill-opacity:0.3; stroke:#000000; stroke-width:0.25;',
          },
        },
      },
    })
    .end({ prettyPrint: true });
};
